chore/FX 5230/automated data removal (#623)
cobalthq/cobalt-product-public-docs: content/en/Platform Deep Dive/Organization/automated-data-removal.md

---
title: "Automated Data Removal"
linkTitle: "Automated Data Removal"
weight: 40
description: >
  Process of automatically deleting data from the Cobalt Platform based on predefined rules and schedules.
---

{{% pageinfo %}}
When your organization's contract expires, the associated data will be removed after a predefined period of time. This period can be extended based on certain conditions. Learn more about our [privacy policy](https://cobalt.io/terms#privacy).
{{% /pageinfo %}}

> When your contract ends, a 15-day grace period begins during which you will still have access to the platform to [export any necessary data](/platform-deep-dive/organization/your-contract/#download-your-organizations-data). Organization Owners will also receive a notification about this. If the contract is renewed, full access to your account will be restored.

Reviewer: It just came to me - I wasn't sure if we did send an email notification for contract end or the grace period. In this [document](https://zombie.atlassian.net/wiki/spaces/FND/pages/2616164635/Revoke+App+Access+to+Churned+Orgs+Product+Requirements), it says 'No email notifications to start - only in platform notifications.' and i can't find any email templates in Figma or Mandrill either.

Author: Good catch, I forgot it's only 60 days before data removal that the first notification goes out. Will remove this sentence.

{{% image src="/deepdive/ContractGateAfterGracePeriod.png" alt="contract gate after grace period" class="mt0" %}}

> After the grace period, you will still be able to sign in to the platform but will no longer have access to your organization's data. However, this data will remain in our system for 90 days after the contract ends.

Reviewer 2: > you will still be able to sign in to the platform but will no longer have access to your organization's data

This isn't strictly true. If I'm reading the code correctly, an org owner can navigate to the dashboard from the [contract ended screen](https://github.com/cobalthq/cobalt-app-web/blob/3efeba4bab49cd67559ff2cb4c61dd21efb76ab9/packages/apps/monolith/src/modules/orgs/ContractGate/BlockedScreen/CardComponents/ContractEndedCardBody.js#L23-L28).

Reviewer 3: Good catch.  So they can see a list of pentests but not click into it?  Just to clarify, by 'dashboard', are you referring to the Home page?

@JakobStorm @molfinn - Any reason we are allowing users to navigate past the 'contract ended screen' after they've churned and the grace period has elapsed?  Or was that unintentional?

Reviewer 2: There's nothing on the backend that would prevent a user from accessing their data after the grace period ends. This includes the public API.

{{% image src="/deepdive/ConfirmToRetainData.png" alt="confirm to retain data" class="mt0" %}}

> You can elect to retain your account for an additional 90 days, if confirmed before the end of the initial period. The extension can only be performed once, and must be done by the Organization Owner(s) for your account.

Reviewer 2: ```suggestion
You can elect to retain your account data for an additional 90 days, if confirmed before the end of the initial period. The extension can only be performed once by an Organization Owner.
```

If you wish to have your data removed earlier, please reach out to your Customer Success Manager (CSM).

## Data Subject to Removal

The following data will be removed:

- User accounts (except those associated with additional organizations not affected by data removal)
- Organization's address, name, and logo
- Assets

> - Pentests

Reviewer 4: I would add vulnerabilities, because DAST and attack surface vulnerabilities are not linked to pentests. We could also add "other engagements" (DRA, SCR)

- Jira integrations

## Notification Recipients

The following user roles will receive email notifications about data removal:

- Organization Owners
- Customer Success Managers

## Notification Schedule

> {{% image src="/deepdive/DataRemovalReminder.png" alt="data removal reminder" class="mt0" %}}

Reviewer: Just realized - maybe it's better to show an image of the email reminder that shows the Keep My Data option since that's the version users will see first

<img width="488" alt="InitialEmailReminder" src="https://github.com/user-attachments/assets/36371b9d-6f73-40a6-bf66-5b1cb8f740b3" />

---

> title: "Automated Data Removal"

Author: Yes, will make a link from the 'Your Contract' page to this one.

linkTitle: "Automated Data Removal"
weight: 20
aliases: /platform-deep-dive/collaboration/organization/manage-users/
description: >
  Process of automatically deleting data from the Cobalt Platform based on predefined rules and schedules.
---

{{% pageinfo %}}
When your organization's contract expires, the associated data will be removed after a predefined period of time. This period can be extended based on certain conditions.
{{% /pageinfo %}}

When your contract ends, a 30-day grace period begins during which you will still have access to the platform to export any necessary data. You will also receive a notification about this. If the contract is renewed, the grace period will be lifted.

{{% image src="/deepdive/ContractGateAfterGracePeriod.png" alt="contract gate after grace period" class="mt0" %}}

> After the grace period, access to the platform will be revoked. However, your data will remain in our system for an additional 90 days (termination period). 

Author: > Just a thought - does it still make sense to use the language of 'revoked' access? Users can technically sign in, they just can't see their data...So something like 'After the grace period, you can still sign into the platform but no longer be able to access your data.'

Yes, good point. If I remember correctly, users won't be able to access the organization anymore. Will clarify that.

> The initial 90 days should include the grace period

Oh ok, didn't know that. Will adjust accordingly.
